src/c_api.rs

```diff
@@ -398,7 +398,7 @@ pub unsafe extern "C" fn c2pa_reader_free(reader_ptr: *mut C2paReader) {
 pub unsafe extern "C" fn c2pa_reader_json(reader_ptr: *mut C2paReader) -> *mut c_char {
     let c2pa_reader: Box<C2paReader> = Box::from_raw(reader_ptr);
     let json = c2pa_reader.json();
-    Box::into_raw(c2pa_reader);
+    let _ = Box::into_raw(c2pa_reader);
```

Comment:
If we don't use the returned pointer, do we even need this? [into_raw](https://doc.rust-lang.org/std/boxed/struct.Box.html#method.into_raw) seems to expect we use the returned pointer... Or is it an artifact needed somehow for the C bindings?

Reply:
I see, so it traces back to `reader_ptr` having to be freed by the caller/provider.